typescript/sdk/src/token/adapters/FuelTokenAdapter.ts

```diff
@@ -0,0 +1,184 @@
+import { Contract, Provider, TransactionRequest, Wallet } from 'fuels';
+
+import { Address, Domain, Numberish } from '@hyperlane-xyz/utils';
+
+import { BaseFuelAdapter } from '../../app/MultiProtocolApp.js';
+import warpRouteAbi from '../../fuel-types/abis/Warp-route-abi.json' assert { type: 'json' };
+import { MultiProtocolProvider } from '../../providers/MultiProtocolProvider.js';
+import { ChainName } from '../../types.js';
+import { TokenMetadata } from '../types.js';
+
+import {
+  IHypTokenAdapter,
+  ITokenAdapter,
+  InterchainGasQuote,
+  TransferParams,
+  TransferRemoteParams,
+} from './ITokenAdapter.js';
+
+// Interacts with Native Fuel token
+export class FuelNativeTokenAdapter
+  extends BaseFuelAdapter
+  implements ITokenAdapter<TransactionRequest>
+{
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider<any>,
```

Comment:
I think we can remove any here
```suggestion
    public readonly multiProvider: MultiProtocolProvider,
```